### PR TITLE
Feature/issue 108 increase test coverage child

### LINKED ISSUE #108: 100% Test Coverage
**Description**
As per today, the test coverage is: **97,3%**, the purpose of this task is to reach **100%**.

**How the coverage is calculated**
The plugin in charge of the test coverage calculation are **[Jacoco](https://github.com/jacoco/jacoco)** along with [SonarCloud](https://sonarcloud.io/dashboard?id=BULL). 
Jacoco it's executed during each build and produces a report inside the target folder of each module.
You can find it at:
`[module-name]/target/site/jacoco`

instead, the SonarCloud report is updated during each release

**The purpose**
The purpose is to identify all the "not covered" code areas and implement specific Unit Test for covering them.

**Additional context**
There are classes and packages excluded from the coverage calculation, such as enum, Java Bean, etc. The whole list of the excluded items are:
* [SonarCloud](https://github.com/HotelsDotCom/bull/blob/master/pom.xml#L73)
* [Jacoco](https://github.com/HotelsDotCom/bull/blob/master/pom.xml#L464)

### PR DESCRIPTION
**Pull request by leharb**
# Pull Request Template

## Description

Please include a summary of the change or which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes #108 

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Please also note any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

## Checklist:

- [ ] The branch follows the best practices naming convention:
     - Use grouping tokens (words) at the beginning of your branch names.
        * `feature`: Feature I'm adding or expanding
        * `bug`: Bug fix or experiment
        * `wip`: Work in progress; stuff I know won't be finished soon
- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules
- [x] My changes have no bad impacts on performances
- [x] My changes have been tested through Unit Tests and the overall coverage has not decreased. Check the coverall report for your branch: [here](https://coveralls.io/github/HotelsDotCom/bull)
- [ ] Any implemented change has been added in the [CHANGELOG.md](./CHANGELOG.md) file 
- [ ] Any implemented change has been added in the [CHANGELOG-JDK8.md](./CHANGELOG-JDK8.md) file 
- [ ] My changes have been applied on a separate branch too with compatibility with Java 8. Follow this [instructions](https://github.com/HotelsDotCom/bull/blob/master/RELEASE.md#3-prepare-the-jdk8-release) for help.

- [ ] The maven version has been increased.

### CODE REVIEW

File: bull-bean-transformer/src/test/java/com/hotels/beans/transformer/ImmutableObjectTransformationTest.java [hunk 1]
```diff
@@ -500,6 +513,33 @@ public void testTransformRaiseAnExceptionIfTheTransformerFunctionIsNotValid() {
 
     }
 
+    /**
```
Comment (fborriello) [inline bull-bean-transformer/src/test/java/com/hotels/beans/transformer/ImmutableObjectTransformationTest.java]: As this test refers to a Mutable object, please move it in the class: `MutableObjectTransformationTest`

File: bull-bean-transformer/src/test/java/com/hotels/beans/transformer/ImmutableObjectTransformationTest.java [hunk 2]
```diff
@@ -500,6 +513,33 @@ public void testTransformRaiseAnExceptionIfTheTransformerFunctionIsNotValid() {
...
+     * Test that an {@link InvalidBeanException} is raised if the class instantiation fails when the method {@code injectValues()} is called.
+     * @throws Exception if something goes wrong
+     */
+    @Test
+    @SuppressWarnings("unchecked")
```
Comment (fborriello) [inline bull-bean-transformer/src/test/java/com/hotels/beans/transformer/ImmutableObjectTransformationTest.java]: i think this is not needed any more

File: bull-bean-transformer/src/test/java/com/hotels/beans/transformer/ImmutableObjectTransformationTest.java [hunk 3]
```diff
@@ -500,6 +513,33 @@ public void testTransformRaiseAnExceptionIfTheTransformerFunctionIsNotValid() {
...
+        Constructor<MutableToFooSimple> constructor = classUtils.getAllArgsConstructor(MutableToFooSimple.class);
+        when(classUtils.getInstance(constructor)).thenThrow(InvalidBeanException.class);
+        when(classUtils.getConstructorParameters(constructor)).thenReturn(new Parameter[] {});
+        when(classUtils.areParameterNamesAvailable(constructor)).thenReturn(true);
+        doReturn(expectedException).when(underTestMock).handleInjectionException(any(), any(), any(), any(), any(), anyBoolean(), any());
+        setField(underTestMock, CLASS_UTILS_FIELD_NAME, classUtils);
+        Method injectValuesMethod =
+            TransformerImpl.class.getDeclaredMethod(INJECT_VALUES_METHOD_NAME, Object.class, Class.class, Constructor.class, String.class, boolean.class);
+        injectValuesMethod.setAccessible(true);
+        //WHEN
```
Comment (fborriello) [inline bull-bean-transformer/src/test/java/com/hotels/beans/transformer/ImmutableObjectTransformationTest.java]: please add an empty line before "//WHEN" and "//THEN"

File: bull-bean-transformer/src/test/java/com/hotels/beans/transformer/ImmutableObjectTransformationTest.java (hunk 1 above)
Comment (leharb) [reply bull-bean-transformer/src/test/java/com/hotels/beans/transformer/ImmutableObjectTransformationTest.java]: This is done.

File: bull-bean-transformer/src/test/java/com/hotels/beans/transformer/ImmutableObjectTransformationTest.java (hunk 2 above)
Comment (leharb) [reply bull-bean-transformer/src/test/java/com/hotels/beans/transformer/ImmutableObjectTransformationTest.java]: This is done.

File: bull-bean-transformer/src/test/java/com/hotels/beans/transformer/ImmutableObjectTransformationTest.java (hunk 3 above)
Comment (leharb) [reply bull-bean-transformer/src/test/java/com/hotels/beans/transformer/ImmutableObjectTransformationTest.java]: This is done.

**Pull request merged**